torch/csrc/Generator.cpp

```diff
@@ -64,7 +79,15 @@ static PyObject * THPGenerator_getState(THPGenerator *self)
   using namespace torch::autograd;
   HANDLE_TH_ERRORS
   Variable var = torch::empty({0}, at::device(at::kCPU).dtype(at::kByte));
+#ifdef USE_CUDA
+  if (self->cdata->device().type() == at::kCPU) {
+    THByteTensor_getRNGState(self->cdata, (THByteTensor*)(var.unsafeGetTensorImpl()));
+  } else {
+    THCRandom_getRNGState(self->cdata, (THByteTensor*)(var.unsafeGetTensorImpl()));
+  }
+#else
```

Comment:
Nit: you'd end up with less code duplication if you structure the ifdef like this:

```
  if (self->cdata->device().type() == at::kCPU) {
    THByteTensor_getRNGState(self->cdata, (THByteTensor*)(var.unsafeGetTensorImpl()));
  } else {
#ifdef USE_CUDA
    TORCH_INTERNAL_ASSERT(self->cdata->device().type() == at::kCUDA);
    THCRandom_getRNGState(self->cdata, (THByteTensor*)(var.unsafeGetTensorImpl()));
#else 
    TORCH_INTERNAL_ASSERT(false, "PyTorch not compiled with CUDA");
#endif 
  }
```

There are some other benefits too: CPU and CUDA aren't the only devices in town. Right now, you'll just fall through to the CUDA code instead of giving a better error message. This is not terribly high priority to fix for this PR, but it should be done properly eventually.

I wasn't originally planning to comment on this, but I saw this macro pattern show up in a bunch of places.